### PR TITLE
docs: Document Tokenizer training Pipeline

### LINKED ISSUE #299: Model  Documentation:  Training Tokenizer

### PR DESCRIPTION
**Pull request by MoosaTae**
## Why this PR
Why we need this PR?

## Changes
- Write some changes here

## Related Issues
Close #299 

## Checklist
- [ ] PR should be in the [Naming convention](../../docs/PR_NAMING.md)
- [ ] Assign yourself in to Assigneees
- [ ] Tag related issues
- [ ] Constants name should be ALL_CAPITAL, function name should be snake_case, and class name should be CamelCase
- [ ] complex function/algorithm should have [Docstring](https://peps.python.org/pep-0257/)
- [ ] 1 PR should not have more than 200 lines changes (Exception for test files). If more than that please open multiple PRs
- [ ] At least PR reviewer must come from the task's team (model, eval, data)

### CODE REVIEW

File: src/model/scripts/spm_training/README.md [hunk 1]
```diff
@@ -1,6 +1,28 @@
...
+## Description
+
+The OpenThaiGPT tokenizer training pipeline is a pipeline for training tokenizer specially for Thai language.
+
+## how to train
+
+1. you need to config argument first [configuration_example/spm/training_v1.yaml](../../configuration_example/spm/training_v1.yaml)
+
+   - output_path (str): The path and prefix to use when saving the trained tokenizer.
+   - vocab_size (int): The size of the vocabulary to use when training the tokenizer. Defaults to the number of available CPU cores.
```
Comment (new5558) [inline src/model/scripts/spm_training/README.md:12]: ` Defaults to the number of available CPU cores.` Why?

File: src/model/scripts/spm_training/README.md [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+   - is_slurm (bool): Whether the code is running on a Slurm cluster. Defaults to False.
```
Comment (new5558) [inline src/model/scripts/spm_training/README.md:13]: What will happened if we set is_slurm = True, False?

File: src/model/scripts/spm_training/README.md [hunk 3]
```diff
@@ -1,6 +1,28 @@
...
+   - load_dataset_path (str): The name of the Hugging Face dataset to load. Defaults to "oscar".
+   - load_dataset_name (str): The name of the dataset split to use. Defaults to "unshuffled_deduplicated_th".
+   - load_dataset_local_path (str): The path to a local directory containing the input data. If specified, the Hugging Face dataset is not used. Defaults to None.
+   - load_dataset_data_type (str): The file type of the input data if using a local directory
+   - large_corpus (bool): Whether to use a large corpus. Defaults to False.
```
Comment (new5558) [inline src/model/scripts/spm_training/README.md:18]:
```suggestion
   - large_corpus (bool): (SPM argument) Whether to use a large corpus. Defaults to False.
```

**Review by boss-chanon — APPROVED** [review]

**Review by new5558 — APPROVED** [review]

File: src/model/scripts/spm_training/README.md [hunk 4]
```diff
@@ -1,6 +1,28 @@
...
+   - mode(spm | bpe): type of tokenizer
```
Comment (new5558) [inline src/model/scripts/spm_training/README.md:19]:
```suggestion
   - mode(spm | bpe): (SPM argument) type of tokenizer
```

Comment (new5558) [reply src/model/scripts/spm_training/README.md:19]: Is there default?

**Review by ArthurMinovsky — APPROVED** [review]

File: src/model/scripts/spm_training/README.md (hunk 1 above)
Comment (MoosaTae) [reply src/model/scripts/spm_training/README.md:12]: I follow in spm_trainer.py document should I change default of it?

File: src/model/scripts/spm_training/README.md (hunk 2 above)
Comment (MoosaTae) [reply src/model/scripts/spm_training/README.md:13]: It is streaming of dataset I will change is_slurm to streaming

File: src/model/scripts/spm_training/README.md (hunk 4 above)
Comment (MoosaTae) [reply src/model/scripts/spm_training/README.md]: default is SPM_MODE, done

File: src/model/configuration_example/spm/README.md [hunk 5]
```diff
@@ -2,10 +2,10 @@
...
+- streaming: streaming mode
 - load_dataset_path: dataset name for load from huggingface don't need if have load_dataset_local_path
 - load_dataset_name: split of dataset use for load_dataset_path
 - load_dataset_local_path: path to load dataset from local if null will use load_dataset_path instead
-- load_dataset_data_type: format type of dataset (ex: json, csv) if null will be huggingface format 
+- load_dataset_data_type: format type of dataset (ex: json, csv) if null will be huggingface format
 - large_corpus: use true when want to train large dataset
-- mode: mode of train tpkenizer sentencepiece (spm: sentencepiece, bpe: byte pair encoding)
\ No newline at end of file
+- mode: mode of train tpkenizer sentencepiece (spm: sentencepiece, bpe: byte pair encoding)
```
Comment (kwankoravich) [inline src/model/configuration_example/spm/README.md:11]: train tokenizer?

**Review by kwankoravich — COMMENTED** [review]
LGTM

**Pull request merged**